### DISCUSSION TITLE
It would be helpful to expose grid snapping options to editor plugins

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#6446] [Editor]
I'm designing an editor plugin and would like to give my users an efficient way of setting the 3D grid snapping size.  The  current process requires you to press Transform button at the top of the editor, scroll down to Configure Snap and then type in the snapping distance.  I want to add a simple OptionButton to the editor toolbar so the user can just click on the snapping size they want.

Unfortunately, I can find no method to set the snapping size in code.  I thought the EditorSettings might do that, but the snapping size seems to be stored elsewhere.

**Comment by Calinou** [comment]
3D grid snapping needs to be reworked to be more similar to 2D in general. Right now, there is no way to adjust the base grid size and snap settings are set independently from the grid size, unlike 2D. There should still be a dialog to configure snapping, but it should configure the grid size instead (and also which grid planes are visible). Snap distance would be configured automatically based on the grid size.

This also allows adding keyboard shortcuts to halve/double the grid size, and therefore the snapping distance.
